homeassistant/components/climate/zwave.py

```diff
@@ -83,45 +83,52 @@ def __init__(self, value, temp_unit):
     def update_properties(self):
         """Callback on data changes for node values."""
         # Operation Mode
-        for value in self._node.get_values(
-                class_id=zwave.const.COMMAND_CLASS_THERMOSTAT_MODE).values():
-            self._current_operation = value.data
-            self._operation_list = list(value.data_items)
-            _LOGGER.debug("self._operation_list=%s", self._operation_list)
-            _LOGGER.debug("self._current_operation=%s",
-                          self._current_operation)
+        self._current_operation = self.get_value(
+            class_id=zwave.const.COMMAND_CLASS_THERMOSTAT_MODE, member='data')
+        operation_list = self.get_value(
+            class_id=zwave.const.COMMAND_CLASS_THERMOSTAT_MODE,
+            member='data_items')
+        if operation_list:
+            self._operation_list = list(operation_list)
+        _LOGGER.debug("self._operation_list=%s", self._operation_list)
+        _LOGGER.debug("self._current_operation=%s", self._current_operation)
+
         # Current Temp
-        for value in (
-                self._node.get_values(
-                    class_id=zwave.const.COMMAND_CLASS_SENSOR_MULTILEVEL)
-                .values()):
-            if value.label == 'Temperature':
-                self._current_temperature = round((float(value.data)), 1)
```

Comment:
Yes, rounding is already done here:https://github.com/turbokongen/home-assistant/blob/f44c55d1012a7b5eb5058197b21c271d50e79e68/homeassistant/components/climate/__init__.py#L697